Projects/Server/GameCounter.h

```diff
@@ -0,0 +1,43 @@
+#ifndef GAMECOUNTER_H
+#define GAMECOUNTER_H
+
+#include "State.h"
+#include "GameHandler.h"
+
+#include <map>
+
+
+
+class GameCounter {
+public:
+	GameCounter(const AbstractPlayer::Ptr& player0, const AbstractPlayer::Ptr& player1, const size_t& gameSample);
```

Comment:
The name `gameSample` is misleading here. In _main_ it makes sense to use that word, because it's sampling the battle of two players, but from this class's point of view, it's actually the number of games to run. Please rename this to something depicting that.

Reply:
You can even separate outcomes from `gamesPlayed` and `gamesSample`.

Reply:
`gameSample` renamed as `_totalGamesToBePlayed` and `gamesPlayed` is renamed as `_currentGameCount` to avoid similar names. These are implemented as new data members.